graylog2-server/src/main/java/org/graylog2/storage/versionprobe/VersionProbeLogger.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.storage.versionprobe;
+
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import org.slf4j.Logger;
+
+public class VersionProbeLogger implements VersionProbeListener {
+
+    private final Logger logger;
+
+    public VersionProbeLogger(Logger logger) {
+        this.logger = logger;
+    }
+
+    @Override
+    public void onRetry(long attemptNumber, long connectionAttempts, @Nullable Throwable cause) {
+        if (connectionAttempts == 0) {
```

Comment:
cause could be added to regular logs too, if available

Reply:
I am not sure what you mean here

Reply:
the `Throwable cause` is not used in the log output, but could be, if not null

Reply:
I think this is by design so, we know that there will be some exceptions for sure during the retry, but we don't want to fill logs with stacktraces during initial waiting for the indexer

Reply:
ok, thanks for clarifying